platform/mellanox/docker-syncd-mlnx/ecmp_calculator/ecmp_calc.py

```diff
@@ -0,0 +1,492 @@
+#!/usr/bin/env python3
+
+
+from ecmp_calc_sdk import sx_open_sdk_connection, sx_get_active_vrids, sx_router_get_ecmp_id, \
+                          sx_router_operational_ecmp_get, sx_get_router_interface, \
+                          sx_port_vport_base_get, sx_router_neigh_get_mac, sx_fdb_uc_mac_addr_get, \
+                          sx_lag_port_group_get, sx_make_ip_prefix_v4, sx_make_ip_prefix_v6, \
+                          sx_vlan_ports_get, sx_ip_addr_to_str, PORT, VPORT, VLAN, SX_ENTRY_NOT_FOUND
+from port_utils import sx_get_ports_map, is_lag
+from packet_scheme import PACKET_SCHEME
+import json, jsonschema
+import argparse
+import ipaddress
+import re
+import subprocess
+import pprint
+import sys
+
+IP_VERSION_IPV4 = 1
+IP_VERSION_IPV6 = 2
+PORT_CHANNEL_IDX = 1
+VRF_NAME_IDX = 1
+IP_VERSION_MAX_MASK_LEN = {IP_VERSION_IPV4: 32, IP_VERSION_IPV6: 128}
+
+INTF_TABLE = 'INTF_TABLE'
+HASH_CALC_PATH = '/usr/bin/sx_hash_calculator'
+HASH_CALC_INPUT_FILE = "/tmp/hash_calculator_input.json"
+HASH_CALC_OUTPUT_FILE = "/tmp/hash_calculator_output.json"
+
+def check_output_pipe(cmd0, *args):
```

Comment:
@liorghub why it so complicated? Why not to use `subprocess.check_output()` and have all the command formatted in a single string? (e.g., "/usr/bin/redis-cli -n 0 keys '\*VRF\*'")

Reply:
I did it to support command with pipeline+grep but there is no real need for that since I can parse the returned string instead of using grep. I changed that.